# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webidl/WorkerNavigator.webidl
# Commit: 244df402d6eb
# Full Hash: 244df402d6ebc15d6845c4144245511021408835
# Author: Andrew Sutherland <asutherland@asutherland.org>
# Date: 2024-10-24 09:44:34
# Regressor Bug: 1131324
# File Overlap Count: 1
# Description:
#   Bug 1131324 - Expose ServiceWorkerContainer on WorkerNavigator. r=dom-worker-reviewers,webidl,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D213725
# ==============================================================================

diff -r 57f70ba5f27e -r 244df402d6eb dom/webidl/WorkerNavigator.webidl
--- a/dom/webidl/WorkerNavigator.webidl	Thu Oct 24 03:02:39 2024 +0000
+++ b/dom/webidl/WorkerNavigator.webidl	Thu Oct 24 03:02:40 2024 +0000
@@ -35,9 +35,15 @@
 // https://gpuweb.github.io/gpuweb/#navigator-gpu
 WorkerNavigator includes NavigatorGPU;
 
-
 // https://w3c.github.io/permissions/#webidl-2112232240
 [Exposed=(Worker)]
 partial interface WorkerNavigator {
   [SameObject] readonly attribute Permissions permissions;
 };
+
+// https://w3c.github.io/ServiceWorker/#navigator-serviceworker
+[Exposed=Worker]
+partial interface WorkerNavigator {
+  [Func="ServiceWorkersEnabled", SameObject]
+  readonly attribute ServiceWorkerContainer serviceWorker;
+};